src/ip/Address.cc

```diff
@@ -695,7 +695,7 @@ Ip::Address::InitAddr(struct addrinfo *&ai)
     // remove any existing data.
     if (ai->ai_addr) delete ai->ai_addr;
 
-    ai->ai_addr = (struct sockaddr*)new sockaddr_in6;
+    ai->ai_addr = (struct sockaddr*)new sockaddr_storage;
```

Comment:
Please use C++ casting
```suggestion
    ai->ai_addr = reinterpret_cast<struct sockaddr *>(new sockaddr_storage);
```